src/main/java/school/hei/haapi/authentication/AwsCognitoJwtIdToken.java

```diff
@@ -0,0 +1,67 @@
+package school.hei.haapi.authentication;
+
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.proc.ConfigurableJWTProcessor;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.Collection;
+import java.util.List;
+
+@Component
+public class AwsCognitoJwtIdToken {
+  @Autowired private JwtConfiguration jwtConfiguration;
+  @Autowired private ConfigurableJWTProcessor configurableJWTProcessor;
+  private static final Log logger = LogFactory.getLog(AwsCognitoJwtIdToken.class);
+
+  private Collection<GrantedAuthority> getAuthorities() {
+    return List.of(
+        new SimpleGrantedAuthority("ROLE_USER"), new SimpleGrantedAuthority("ROLE_ADMIN"));
```

Comment:
On second thought, use Cognito for authentication only (so no role to add in the token it sends). The corresponding role will be stored by the API in its own database.

Reply:
So the users informations are duplicated in the API database ? including the id ?
Because if I understand your "second thought", we will use the id included in the token then the "local User" by using id to identify the role ?

Reply:
Yeah, duplicated. This is the notion of primary source: we use the API's database as main source of information. And Cognito is just a sort of replica. Note that identifiers are different between API's db and Cognito's db.

Also note that the role is not forwarded to Cognito, only the user's [first]name and email

Reply:
So the primary key to identify a user is the unique ([first]name,email) ?

Reply:
no, there should be an additional uuid. anyway, that should be the responsibility of cognito. look at its api.

Reply:
In our case I just used the email as we said.